dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java

```diff
@@ -23,95 +18,51 @@ public class CacheConfigManager {
 
     private static final CacheConfigManager CACHE_CONFIG_MANAGER = new CacheConfigManager();
 
-    private static LoadingCache<String, String> installationAccessTokenCache;
+    private static volatile LoadingCache<Long, GitHub> githubClientAPICache;
```

Comment:
Recommend that the fields and methods of `CacheConfigManager` be non-`static` (except the field storing the reference to the singleton and the method that retrieves it), giving us the same functionality but with more flexibility later (to do things like instantiate multiple `CacheConfigManagers` for testing or composition or whatever).  For example:

```
public class CacheConfigManager {
    private static final CacheConfigManager CACHE_CONFIG_MANAGER = new CacheConfigManager();
    private volatile LoadingCache<Long, GitHub> githubClientAPICache;
    public static CacheConfigManager getInstance() {
        return CACHE_CONFIG_MANAGER;
    }
    public void initCache(String githubAppId, String gitHubPrivateKeyFile) {
        [...]
    }
}
```
Then, to use:
```
    CacheConfigManager.getInstance().initCache([...]);
```

Reply:
Regarding `volatile`, it only helps to ensure the consistency of the volatile field's value itself, the referenced data structure (the actual cache object and data within) is not affected.  If multiple threads manipulate an object at the same time, a good way to ensure consistency is to `synchronize` so the happens-before relationships are correct.  A good reference:  https://docs.oracle.com/javase/specs/jls/se18/html/jls-17.html#jls-17.4.5

Reply:
I think I'll leave it till we have a need for extra caches to engineer that, we may prefer a different class name (rather than multiple caches in the same class) or other changes that I think make this premature